assets/blocks/editor-components/toolbar-dropdown/index.js

```diff
@@ -24,62 +29,88 @@ import {
  * Dropdown for the editor toolbar.
  *
  * @param {Object}           props
- * @param {DropdownOption[]} props.options        Dropdown options.
- * @param {string}           [props.optionsLabel] Options label.
- * @param {Object}           props.icon           Icon for the toolbar.
- * @param {string}           props.value          Current dropdown value.
- * @param {Function}         props.onChange       Dropdown change callback, which receive
- *                                                the new value as argument.
+ * @param {DropdownOption[]} props.options             Dropdown options.
+ * @param {string}           [props.optionsLabel]      Options label.
+ * @param {Object}           props.icon                Icon for the toolbar.
+ * @param {string}           props.value               Current dropdown value.
+ * @param {Function}         props.onChange            Dropdown change callback, which receive the new value as argument.
+ * @param {Object}           props.toggleProps         Props passed to the toggle element.
+ * @param {Object}           props.popoverProps        Props passed to the popover component.
+ * @param {Function}         props.renderMenuItemProps Render function for a menu item. Should return a props object.
  */
 const ToolbarDropdown = ( {
 	options,
 	optionsLabel,
 	icon,
 	value,
 	onChange,
+	toggleProps,
+	renderMenuItemProps,
+	popoverProps,
+	...props
 } ) => {
 	const selectedOption = options.find( ( option ) => value === option.value );
 
 	return (
 		<Dropdown
-			popoverProps={ { isAlternate: true } }
+			className="sensei-toolbar-dropdown"
+			popoverProps={ {
+				isAlternate: true,
+				position: 'bottom right left',
+				focusOnMount: true,
+				...popoverProps,
+				className: classnames(
+					popoverProps?.className,
+					'sensei-toolbar-dropdown__popover'
+				),
+			} }
 			renderToggle={ ( { isOpen, onToggle } ) => (
 				<Button
 					onClick={ onToggle }
 					icon={ icon }
 					aria-expanded={ isOpen }
 					aria-haspopup="true"
-				>
-					{ selectedOption.label }
-				</Button>
+					{ ...toggleProps }
+					children={
+						toggleProps?.children
+							? toggleProps?.children( selectedOption )
```

Comment:
Is optional chaining needed here? Also I think that it is a bit confusing that we use children as the name of a callback. Maybe something like `getToggleChildren` fits better?

Reply:
This is a common pattern in React, to have a callback function as the children if they are rendered in a dynamic way. Like in `<Dropdown> { ( onClose ) => ... }</DropDown>`, that render function will be the `children` prop. 
Since the `toggleProps` parameter is the one passed to this component, it makes sense to also have it's render function be a part of that instead of a separate prop.
Optional chaining is there because `toggleProps` is optional and could be null.

Reply:
Yep. What I am trying to say is that the `toggleProps?.children( selectedOption )` expression will only get evaluated if `toggleProps?.children` is true so the second one is not needed right?

Reply:
Oh right. Updated here: cbcfe2e